File: day3/Cognizant GenC.py
def check_lucky_number():
    # Read the car number input from the user
    car_no = input("Enter the car no:")
    
    # Check if the string consists of exactly 4 digits
    if len(car_no) != 4 or not car_no.isdigit():
        print(f"{car_no} is not a valid car number")
        return
        
    # Safeguard against the "0000" edge case as problem notes "positive numbers"
    if int(car_no) <= 0:
        print(f"{car_no} is not a valid car number")
        return

    # Calculate the sum of the digits
    digit_sum = sum(int(digit) for digit in car_no)

    # Check if the sum is divisible by 3, 5, or 7
    if digit_sum % 3 == 0 or digit_sum % 5 == 0 or digit_sum % 7 == 0:
        print("Lucky Number")
    else:
        print("Sorry its not my lucky number")

# Run the function
if __name__ == "__main__":
    check_lucky_number()

File: day3/Cognizant GenC.cpp
#include <iostream>
#include <string>
#include <cctype>

using namespace std;

bool isValidDigits(const string& str) {
    for (char const &c : str) {
        if (isdigit(c) == 0) return false;
    }
    return true;
}

int main() {
    string car_no;
    cout << "Enter the car no:";
    cin >> car_no;
    
    if (car_no.length() != 4 || !isValidDigits(car_no)) {
        cout << car_no << " is not a valid car number" << endl;
        return 0;
    }
    if (stoi(car_no) <= 0) {
        cout << car_no << " is not a valid car number" << endl;
        return 0;
    }
    int digit_sum = 0;
    for (char c : car_no) {
        digit_sum += (c - '0');
    }
    
    if (digit_sum % 3 == 0 || digit_sum % 5 == 0 || digit_sum % 7 == 0) {
        cout << "Lucky Number" << endl;
    } else {
        cout << "Sorry its not my lucky number" << endl;
    }
    
    return 0;
}
